pages/governance/proposal/index.governance.tsx

```diff
@@ -19,34 +17,27 @@ export default function DynamicProposal() {
   const [proposal, setProposal] = useState<CustomProposalType>();
   const [ipfs, setIpfs] = useState<IpfsType>();
 
-  async function updateProposal() {
-    const { values, ...rest } = await governanceContract.getProposal({ proposalId: id });
-    setProposal(await enhanceProposalWithTimes(rest));
+  async function initialize() {
+    try {
+      const { values, ...rest } = await governanceContract.getProposal({ proposalId: id });
+      const proposal = await enhanceProposalWithTimes(rest);
+      setProposal(proposal);
+      const newIpfs = {
+        id,
+        originalHash: proposal.ipfsHash,
+        ...(await getProposalMetadata(proposal.ipfsHash, governanceConfig?.ipfsGateway)),
+      };
+      setIpfs(newIpfs);
+    } catch (e) {
+      console.log(e);
+      setTimeout(initialize, 5000);
+    }
   }
 
-  async function fetchIpfs() {
-    if (!proposal) return;
-    const newIpfs = {
-      id,
-      originalHash: proposal.ipfsHash,
-      ...(await getProposalMetadata(proposal.ipfsHash, governanceConfig?.ipfsGateway)),
-    };
-    setIpfs(newIpfs);
-  }
-
-  // poll every 10s
-  usePolling(
```

Comment:
this polling effectively resulted in double fetching of everything as ProposalPage is polling as well